lib/streamlit/elements/multiselect.py

```diff
@@ -216,14 +216,17 @@ def _check_and_convert_to_indices(
                 else:
                     default_values = list(default_values)
             if len(default_values) != 0 and isinstance(default_values[0], Enum):
-                str_default_values = [str(enum) for enum in default_values]
-                mapped_opt_keys = [str(enum) for enum in opt]
-                for value in str_default_values:
-                    if value not in mapped_opt_keys:
+                for value in default_values:
+                    if value not in opt:
                         raise StreamlitAPIException(
-                            "Every Multiselect default value must exist in options"
+                            """Every Multiselect default value must exist in options. 
```

Comment:
Text suggestion:

> One of the default values you provided to `multiselect` could not be found among the options provided.
>
> If this error seems incorrect, and you are certain that the value should be present in the options, it likely has to do with the behavior of `Enum`s during streamlit reruns. Unfortunately, if the `Enum` is defined in the main script file, comparison breaks for its instances when re-running the script.
> This issue can be mitigated by moving the `Enum` declaration to another module.
>
> Offending default value: {value}

Reply:
Likewise, the non-enum version could read:

> One of the default values you provided to `multiselect` could not be found among the options provided.
>
> Offending default value: {value}

Including the offending value in the error makes it easier to diagnose where you've made a mistake, or, in the `Enum` case, it makes it easier to judge whether you should move your `Enum`, or, your mistake was of the more traditional kind.

Reply:
These are great comments throughout. Much appreciated!